pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -159,7 +159,7 @@
     // The goal is to optimize batch density while also ensuring that a producer never waits longer than the configured
     // batchingMaxPublishDelayMicros to send a batch.
     // Only update from within synchronized block on this producer.
-    private ScheduledFuture<?> batchFlushTask;
+    private Timeout batchFlushTask;
```

Comment:
The corresponding comment above should be updated also.

Reply:
@tisonkun thanks for review, this change not change the logic for when trigger the schecule task. just change the scheduler  to hashwheeltimer for better performance. maybe the comment can remain the same.

Reply:
Thanks for your explanation! SGTM.